sql/core/src/test/scala/org/apache/spark/sql/parquet/ParquetSchemaMergeConfigSuite.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.parquet
+
+import org.apache.spark.sql.{SQLConf, QueryTest}
+import org.apache.spark.sql.test.TestSQLContext
+import org.scalatest.BeforeAndAfterAll
+
+case class ParquetData2(int2Field: Int, string2Field: String)
+
+
+class ParquetSchemaMergeConfigSuite  extends QueryTest with ParquetTest with BeforeAndAfterAll {
+  val sqlContext = TestSQLContext
+  val defaultPartitionName = "__HIVE_DEFAULT_PARTITION__"
+  case class ParquetData2(int2Field: Int, string2Field: String)
+
+  private val originalConf = sqlContext.conf
+                          .getConf(SQLConf.PARQUET_MERGE_SCHEMA_ENABLED)
```

Comment:
`ParquetTest` extends `SQLTestUtils`, which provides a bunch of convenient utility methods.  One of them is `withSQLConf`, which is used to deal with temporary configuration changes, so that you don't need to restore original configurations by hand. For example:

```scala
test("a fancy test") {
  withSQLConf(SQLConf.PARQUET_SCHEMA_MERGING_ENABLED.key, "false") {
    // Do the test
  }
}
```

With this you don't need `BeforeAndAfterAll` for this test suite.